test/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -3123,4 +3123,94 @@ public void concatenatedAfterMissingGZIPTrailer() throws Exception {
         assertEquals(1, oversizedCounter.get());
         nextExpect(null);
     }
+
+    @Test
+    public void incompleteContainerNonContinuable() throws Exception {
```

Comment:
I'm aware of this naming convention, but I've been ambivalent about it personally. I've viewed it as a (perhaps small) cognitive barrier to writing tests, and I'm not sure I'd actually rely on the name of a failing test to reliably tell me what was wrong, even if it followed this convention. I think I'd still click through to the line in the test that resulted in the failure and build context from there, then jump into debugging.

I do agree that including something about the expectation in the name is beneficial, but I typically only do this when the expectation is something other than "succeeds" or "doesn't throw", since most tests fall under that category. If I have a test that asserts failure I (hopefully always) suffix it with "Fails".

I did enjoy, for the brief time I wrote Scala, being able to name tests using free text. It also felt more natural to do per-test parameterization and nesting. If we can come up with something that feels close to that, I'd be all for it.

Let's document some conventions for test naming and structure (using JUnit 5), then revisit existing tests to ensure they match those conventions. See #594